src/runtime_src/core/edge/user/device_linux.cpp

```diff
@@ -216,14 +218,28 @@ struct aie_core_info_sysfs
   {
     boost::property_tree::ptree ptarray;
     aie_metadata_info aie_meta = get_aie_metadata_info(device);
-    const std::string aiepart = std::to_string(aie_meta.shim_row) + "_" + std::to_string(aie_meta.num_cols);
-    const aie_sys_parser asp(aiepart);
-
-    /* Loop each all aie core tiles and collect core, dma, events, errors, locks status. */
-    for (int i = 0; i < aie_meta.num_cols; ++i)
-      for (int j = 0; j < (aie_meta.num_rows-1); ++j)
-        ptarray.push_back(std::make_pair(std::to_string(i) + "_" + std::to_string(j),
-                          asp.aie_sys_read(i,(j + aie_meta.core_row))));
+    std::string base_path = "/sys/class/aie/";
+    std::regex pattern(R"(aiepart_(\d+)_(\d+))");
+    int start_col=0, num_col=0;
+    for (const auto& entry : std::filesystem::directory_iterator(base_path)) {
+        if (entry.is_directory()) {
+            std::string dir_name = entry.path().filename().string();
```

Comment:
Please use early exit to avoid deep indentation.
```suggestion
        if (!entry.is_directory())  
          continue;
```